src/data/scripts/campaign/colonies/VayraColonialManager.java

```diff
@@ -585,27 +671,29 @@ public void pickNextUpgrade(MarketAPI market) {
             }
         }
 
-        if (upgrade != null && market.hasIndustry(upgrade)) {
-            Industry ind = market.getIndustry(upgrade);
-            if (ind != null) {
-                ind.startUpgrading();
-                log.info(String.format("upgrading %s on %s", upgrade, market.getName()));
-            } else {
-                log.error(String.format("[ERROR] WANTED TO UPGRADE INDUSTRY %s on %s BUT COULDN'T BECAUSE IT WAS NULL", upgrade, market.getName()));
+        if (!AOTD_ENABLED) {
```

Comment:
also, make a LunaSetting for this, so it can be turned off via LunaSettings for "just in case" since this part seems to be problematic and I don't want to introduce a hard-dependancy on AOTD mod ID.

The setting should be "Disable L'Interstellaire upgrades" (defaulting to false) and this line should end up being
```
if (!AOTD_ENABLED) {  // Never perform upgrades if AOTD is enabled, regardless of upgrades luna setting
    if (!UPGRADES_DISABLED) {  // User-customizable upgrades disabling part
        /*  upgrade code */
    } else {
       log.info("Not performing l'interstellaire upgrades because it's been turned off in luna settings");
    }
} else {
    log.info("Not performing l'interstellaire upgrades because AOTD is turned on");
}
```

so that upgrades don't happen if AOTD is enabled or upgrades are disabled - in other words, upgrades should happen if AOTD is disabled and it's not been disabled by the user in the luna settings

Reply:
at first I thought something like this could be more useful
```
boolean upgradesEnabled = !AOTD_ENABLED && !UPGRADES_DISABLED;  // ('true' for AOTD being disabled and 'true' for upgrades not being disabled)
if (upgradesEnabled) { 
    /* upgrade code */
}
```
but the double if give me better way of differentiating and logging the actual reason of why they're not being performed.

especially because I wanted to say
```
boolean upgradesEnabled = !AOTD_ENABLED || !UPGRADES_DISABLED;
```
and that could still enable upgrades even if AOTD was enabled. :D

Reply:
done. testing.

Reply:
resolving